components/alert/alert-toast.js

```diff
@@ -256,19 +332,34 @@ class AlertToast extends LitElement {
 				}
 			}
 			this.setAttribute('role', 'alert');
+			await this.updateComplete;
+			const height = this._innerContainer.offsetHeight;
+			this.dispatchEvent(new CustomEvent(
+				'd2l-alert-toast-open', {
+					bubbles: true,
+					composed: false,
+					detail: { height }
+				}
+			));
 		} else {
+			let height = 0;
+			let bottom = 0;
+			if (this._innerContainer) {
+				height = this._innerContainer.offsetHeight;
+				bottom = parseFloat(getComputedStyle(this._innerContainer).getPropertyValue('bottom'));
+			}
 			if (reduceMotion || this._state === states.PREOPENING) {
 				cancelAnimationFrame(this._preopenFrame);
 				this.removeAttribute('role');
 				this._state = states.CLOSED;
-			} else if (this._state === states.OPENING || this._state === states.OPEN) {
+			} else if (this._state === states.OPENING || this._state === states.OPEN || this._state === states.SLIDING) {
 				this._state = states.CLOSING;
 			}
 			this.dispatchEvent(new CustomEvent(
 				'd2l-alert-toast-close', {
 					bubbles: true,
 					composed: false,
-					detail: { action: this._action }
```

Comment:
This doesn't seem to be hooked up to anything and doesn't seem to be used so I cleaned it up while here.

Reply:
Ah yup that was from earlier work and didn't end up being needed - will remove